### PR TITLE
QuoteAttestation: Add new field for request type

### PR DESCRIPTION
**Pull request by avalluri**
New field 'type' is added to hold the type of attestation request. This is to support initiating the QuoteAttestation from CSR only quote validation. In this case, the quote attestation controller does not proceed with key wrapping.

### CODE REVIEW

File: internal/k8sutil/k8sutil.go [hunk 1]
```diff
@@ -94,6 +94,7 @@ func QuoteAttestationDeliver(
 	ctx context.Context,
 	c client.Client,
 	instanceName, namespace string,
+	requestType tcsapi.QuoteAttestationRequestType,
```
Comment (ipuustin) [inline internal/k8sutil/k8sutil.go:97]: Shouldn't this be `RequestTypeQuoteAttestation`?

**Review by ipuustin — CHANGES_REQUESTED** [review]

File: internal/k8sutil/k8sutil.go (hunk 1 above)
Comment (avalluri) [reply internal/k8sutil/k8sutil.go:97]: No, The argument type name is `QuoteAttestationRequestType`. The possible values are `RequestTypeQuoteAttestation` or `RequestTypeKeyProvisioning`. So, nothing wrong here !. Or did i miss something :(

Comment (ipuustin) [reply internal/k8sutil/k8sutil.go:97]: Ok, I get it now! I was confused with the string permutation for a minute. :-)

**Review by ipuustin — APPROVED** [review]
LGTM!

**Review by irisdingbj — COMMENTED** [review]
Shall we make PublicKey and SignerNames as optional?

**Comment by irisdingbj** [comment]
> > Shall we make PublicKey and SignerNames as optional?
> 
> Make sense to make them optional, in case of simple quote verification, but KMRA does not support quote verification without them.

yes, km-attest requires a pubkey. So at least make SignerNames as optional?

**Comment by avalluri** [comment]
> > Make sense to make them optional, in case of simple quote verification, but KMRA does not support quote verification without them.
> 
> yes, km-attest requires a pubkey. So at least make SignerNames as optional?

What I remember is that their whole logic was tied with the 'SignerName', to validate if a client request (quote verify / key wrap) is valid.
By the way, I made a proposal to move `SingerNames` -> `SignerName`: https://github.com/intel/trusted-certificate-issuer/issues/20. @irisdingbj Would please provide your opinion.

**Comment by avalluri** [comment]
@irisdingbj Can we merge this change?

**Comment by avalluri** [comment]
I am merging this change as it is needed for further changes. Let's take it as a follow-up PR if needs any further API changes.

**Pull request merged**